### PR TITLE
Add API for collision free point/pose IK

### PR DESCRIPTION
**Pull request by PeterMitrano**
This adds dependency on bio IK, for better or for worse
https://github.com/UM-ARM-Lab/bio_ik

### CODE REVIEW

File: jacobian_follower/include/jacobian_follower/jacobian_follower.hpp [hunk 1]
```diff
@@ -144,6 +145,17 @@ class JacobianFollower {
   std::vector<std::vector<double>> compute_IK_solutions(geometry_msgs::Pose target_pose,
                                                         const std::string &group_name) const;
 
+  std::optional<moveit_msgs::RobotState> computeCollisionFreePointIK(
```
Comment (bsaund) [inline jacobian_follower/include/jacobian_follower/jacobian_follower.hpp:148]: There are a lot of arguments. You might consider defining a wrapper class type to reduce number of arguments.

Downside in this case is you'll need a decoder for pybind.
Upside is less likely to make a mistake when passing arguments across from python

File: jacobian_follower/src/jacobian_follower/jacobian_follower.cpp [hunk 2]
```diff
@@ -241,6 +244,104 @@ PlanResult JacobianFollower::plan(JacobianTrajectoryCommand traj_command) {
   return {robot_trajectory, reached_target};
 }
 
+bool isStateValid(planning_scene::PlanningScenePtr planning_scene, moveit::core::RobotState *robot_state,
+                  moveit::core::JointModelGroup const *jmg, double const *joint_positions) {
+  robot_state->setJointGroupPositions(jmg, joint_positions);
+  robot_state->update();  // I *think* this updates the internally stored transforms, and probably isn't necessary
```
Comment (bsaund) [inline jacobian_follower/src/jacobian_follower/jacobian_follower.cpp:250]: Might be worth figuring this out before merging

File: jacobian_follower/include/jacobian_follower/jacobian_follower.hpp [hunk 3]
```diff
@@ -144,6 +145,17 @@ class JacobianFollower {
...
+      moveit_msgs::RobotState const &default_robot_state, const std::vector<geometry_msgs::Point> &target_point,
+      const std::string &group_name, const std::vector<std::string> &tip_names,
+      const moveit_msgs::PlanningScene &scene_smg) const;
+
+  std::optional<moveit_msgs::RobotState> computeCollisionFreePoseIK(const moveit_msgs::RobotState &default_robot_state,
```
Comment (bsaund) [inline jacobian_follower/include/jacobian_follower/jacobian_follower.hpp:153]: What is `tip_names`? Add documentation or change argument name

**Review by bsaund — COMMENTED** [review]
I haven't tried the tutorial, I'm glad it's included though

File: jacobian_follower/include/jacobian_follower/jacobian_follower.hpp (hunk 1 above)
Comment (PeterMitrano) [reply jacobian_follower/include/jacobian_follower/jacobian_follower.hpp:148]: I agree there are benefits to a struct, but pybind already gives pretty nice error messages, so for now I'll leave it as is.

**Pull request merged**